Changes for safety firmware releases >= 0.2.6

mhei opened the pull request:
Please find the required changes for upcoming fw 0.2.6

Note, that the actual firmware binary and the version bump is still missing in this PR.

mhei committed:
fw_info: add parameter block version field

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
param-block: add support for versioned parameter blocks

This adapt the ra-pb-dump and ra-pb-create comand line tools.

While it is still possible to read older parameter block file
and also YAML files with older structure, it is not possible
to create an older, unversioned parameter block anymore.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Drop ra-gen-param-block tool

New parameter blocks are more complex and creating from
command line parameters is... even more complex, so R.I.P.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Extend sample parameter blocks

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Adapt firmware update script to handle new version

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Display new parameter version info field

Starting with firmware version 0.2.6, the firmware presents its expected
and/or understood parameter version in the FW_VERSION message.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Adaptions for 3rd contactor

Dedicated contactor errors were removed from Charge State 1 message,
the new room was used to add support for a 3rd contactor.
Also the status field meaning was slightly adapted.

Note, that there is currently no way anymore to tell upper layers
which exact contactor failed. This will probably re-added in a new
safety controller firmware.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Sync error codes and add new HVSWITCH related ones

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-pb-create: add missing new flag in example usage

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-pb-create: factor out state enum and add debug

While at - important - initialize the variable.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
ra-pb-create: fix wrong initial offset in check for empty document

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Drop contactor error functions

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Bump SOVERSION

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Add extra newline after version field

This aligns the output with the following sequences.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

lategoodbye commented:
I think we should provide some kind of compatibility matrix ra-utils | safety fw | parameter block version in the README file. Because providing this in the changelog is not userfriendly.

mhei committed:
Add safety firmware releases v0.2.6

Changelog since 0.2.5:
- added temperature filtering
- added temperature offset parameter (parameter block)
- added support for 3rd contactor (changed Charge State message)
- added HV connector parameterisation (parameter block)
- parameter blocks are now versioned
- parameter block version can be retrieved at runtime

Known bugs:
- Charge SOM: HV switch error handling is incomplete and does not trigger safe state

WARNING: this version is for chargebyte internal testing/use ONLY
and thus unsupported for the moment.

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
README.md: add compatibility warning

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>

mhei committed:
Bump version to 0.9.0

Signed-off-by: Michael Heimpold <michael.heimpold@chargebyte.com>
